src/core/components/InstallButton/index.js

```diff
@@ -71,6 +70,18 @@ export class InstallButtonBase extends React.Component {
     return i18n.sprintf(label, { name });
   }
 
+  getDownloadProgress() {
```

Comment:
I don't think it's necessary right now to make this architectural change but I was just thinking that we could use redux in a way that still makes the component *dumb*. Instead of accessing redux data with `downloadProgressId` it could be more generic, like `switchId`. Thus, it wouldn't need to be download specific. 

By using redux, I think it allows us to separate the *producers* of data from the *consumers* of data better. The `Addon` component is the one that dispatches download progress data but `Switch` is really the only one that consumes it so I don't see a good reason for `Addon` to consume the data only to pass it down through its children until it reaches `Switch`.